mathlibtools/lib.py

```diff
@@ -145,23 +145,27 @@ def get_mathlib_archive(rev: str, url:str = '', force: bool = False,
     Return the archive Path. Will raise LeanDownloadError if nothing works.
     """
 
-    fname = rev + '.tar.gz'
-    path = DOT_MATHLIB/fname
+    # we check for xz archives first
+    fnames = [rev + '.tar.xz', rev + '.tar.gz']
+    paths = [DOT_MATHLIB/fname for fname in fnames]
     if not force:
         log.info('Looking for local mathlib oleans')
-        if path.exists():
-            log.info('Found local mathlib oleans')
-            return path
+        for path in paths:
+            if path.exists():
+                log.info('Found local mathlib oleans')
+                return path
     log.info('Looking for remote mathlib oleans')
-    try:
-        base_url = url or get_download_url()
-        download(base_url+fname, path)
-        log.info('Found mathlib oleans at '+base_url)
-        return path
-    except LeanDownloadError:
-        pass
+    for fname, path in zip(fnames, paths):
+        try:
+            base_url = url or get_download_url()
+            download(base_url+fname, path)
+            log.info('Found mathlib oleans at '+base_url)
+            return path
+        except LeanDownloadError:
+            pass
     log.info('Looking for GitHub mathlib oleans')
-    download(nightly_url(rev, repo), path)
+    # nightlies will only store gz archives
+    download(nightly_url(rev, repo), paths[1])
```

Comment:
```suggestion
    download(nightly_url(rev, repo), DOT_MATHLIB/(rev + '.tar.gz'))
```
I think this makes the intend clearer, and if more robust against the next iteration when someone will ask to first try yet another compression format.